bucket.py

```diff
@@ -0,0 +1,77 @@
+
+
+class Bucket:
+    """Bins to hold coin values.
+
+    Attributes:
+        _label (str): Bin's label
+        _weight (int): Bin's weight, coin's total weight
+        _coins (object: coin): The coins inside this bin with index as its key
+    """
+
+    def __init__(self, label):
+        self._label = label
+        self._weight = 0
+        self._coins = {}
+
+    def __str__(self):
+        list = [v.value for (k, v) in self.coins.items()]
+        return "Bin: " + str(self.label) + " -- Weight: " + str(self.weight) + " -- Coins: " + str(list)
+
+    def __len__(self):
+        return len(self.coins)
+
+    @property
+    def label(self):
+        """str: Property of bin's label"""
+        return self._label
+
+    @property
+    def coins(self):
+        """:dict:'coin': Property of bin's coins"""
+        return self._coins
+
+    @property
+    def weight(self):
+        """int: Property of the bin's weight, total amount of coin's weights"""
+        return self._weight
+
+    @weight.setter
+    def weight(self, weight):
+        self._weight = weight
+
+    def add_coin(self, coin, index):
+        """Add coin into bin with its index from list as key
+
+        Args:
```

Comment:
They probably won't ding you on this, but technically parameter is a variable in a method definition. When a method is called, the arguments are the data you pass into the method's parameters. Parameter is variable in the declaration of function. Argument is the actual value of this variable that gets passed to function